src/controller/java/src/chip/objecttlv/ObjectToTlv.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package chip.objecttlv
+
+import chip.jsontlv.fromJsonString
+import chip.tlv.AnonymousTag
+import chip.tlv.Tag
+import chip.tlv.TlvWriter
+
+/**
+ * Converts kotlin value into TLV.
+ *
+ * @param value kotlin value to be converted to TLV. If String value is in json format, it is judged
+ *   as json and converted to TLV.
+ */
+fun TlvWriter.fromObject(value: Any?, tag: Tag = AnonymousTag): TlvWriter {
+  when (value) {
+    null -> {
+      putNull(tag)
+    }
+    is Int -> {
+      put(tag, value)
+    }
+    is Long -> {
+      put(tag, value)
+    }
+    is Short -> {
+      put(tag, value)
+    }
+    is Byte -> {
+      put(tag, value)
+    }
+    is UInt -> {
+      put(tag, value)
+    }
+    is ULong -> {
+      put(tag, value)
+    }
+    is UShort -> {
+      put(tag, value)
+    }
+    is UByte -> {
+      put(tag, value)
+    }
+    is Boolean -> {
+      put(tag, value)
+    }
+    is Float -> {
+      put(tag, value)
+    }
+    is Double -> {
+      put(tag, value)
+    }
+    is ByteArray -> {
+      put(tag, value)
+    }
+    is List<*> -> {
+      startArray(tag)
+      value.forEach { fromObject(it) }
+      endArray()
+    }
+    is String -> {
+      try {
+        fromJsonString(value)
```

Comment:
actually I don't think it is valuable to add ObjectToTlv. fromJsonString(value) is only helpful when constructing invoke command payload. it is not immediate helpful to construct the payload for IM write, I mean it needs a little bit more logic to construct the IM payload for write

For write, it would be something
  val dataWithStruct = TlvWriter().fromJsonString(jsonBlob)
  val tlvReader = TlvReader(dataWithStruct)
  tlvReader.skipElement()
  while (!tlvReader.isEndOfTlv())) {
    val elementBytes = TlvWriter().startStructure(AnonymousTag).copyElement(tlvReader).getEncoded()
    val innerTlvReader = TlvReader(elementBytes)
    innerTlvReader.skipElement()
    val tag: ContextSpecificTag? = innerTlvReader.peekElement().tag as? ContextSpecificTag
    val tagNumber = tag?.tagNumber?.toUInt()
    val value = TlvWriter().copyElement(AnonymousTag, innerTlvReader).getEncoded()
 then construct the path and value for IM write API

Reply:
I would expect to add new menu item for Command that can handle json string with a struct of arguments
add new menu item for write that can handle a json string with several attributes inside.

Reply:
please consider to rework this using https://github.com/project-chip/connectedhomeip/issues/28592, thanks